File: provider/vllm.go
package provider

import "fmt"

type VLLMProvider struct {
	BaseURL       string
	APIKey        string
	ModelName     string
	MaxTokens     int
	ContextWindow int
	Temperature   float64
	TopP          float64
}

func NewVLLMProvider(baseURL, apiKey, modelName string, maxTokens, contextWindow int, temperature, topP float64) *VLLMProvider {
	return &VLLMProvider{
		BaseURL:       baseURL,
		APIKey:        apiKey,
		ModelName:     modelName,
		MaxTokens:     maxTokens,
		ContextWindow: contextWindow,
		Temperature:   temperature,
		TopP:          topP,
	}
}

func (p *VLLMProvider) ProviderName() string { return "vllm" }

func (p *VLLMProvider) BuildURL() string {
	return fmt.Sprintf("%s/chat/completions", p.BaseURL)
}

func (p *VLLMProvider) BuildHeaders() map[string]string {
	return map[string]string{
		"Authorization": "Bearer " + p.APIKey,
		"Content-Type":  "application/json",
	}
}

func (p *VLLMProvider) GetContextWindow() int { return p.ContextWindow }
func (p *VLLMProvider) GetMaxTokens() int     { return p.MaxTokens }
func (p *VLLMProvider) GetMaxInput() int      { return p.ContextWindow - p.MaxTokens - 2048 }

File: v1/provider/vllm.py
"""
vLLM Provider —— 对接 vLLM 推理引擎（OpenAI 兼容 API）。

默认端点: http://localhost:8000/v1
"""
from typing import Dict

from provider.base import BaseProvider


class VLLMProvider(BaseProvider):
    """vLLM 推理引擎 Provider"""

    @property
    def provider_name(self) -> str:
        return "vllm"

    def _build_url(self) -> str:
        return f"{self.base_url}/chat/completions"

    def _build_headers(self) -> Dict[str, str]:
        return {
            "Authorization": f"Bearer {self.api_key}",
            "Content-Type": "application/json",
        }
